packages/@aws-cdk/aws-ecs-patterns/lib/base/load-balanced-service-base.ts

```diff
@@ -108,11 +120,21 @@ export abstract class LoadBalancedServiceBase extends cdk.Construct {
 
   public readonly targetGroup: elbv2.ApplicationTargetGroup | elbv2.NetworkTargetGroup;
 
+  public readonly cluster: ecs.ICluster;
+
   public readonly logDriver?: ecs.LogDriver;
 
   constructor(scope: cdk.Construct, id: string, props: LoadBalancedServiceBaseProps) {
     super(scope, id);
 
+    if (props.cluster && props.vpc) {
```

Comment:
Please move this logic to `LoadBalancedFargateService` as discussed.

Reply:
I am not sure how this should work, as the constructor needs the cluster and I cannot do stuff before calling super...

Reply:
you are right. OK, let's leave this here, but make sure that for EC2 cluster is required.

Reply:
And how? The base class does not know about EC2 or Fargate, does it?

Why shouldn't it be possible to do:

```
const service = new LoadBalanced....;
service.cluster.addCapacity(...);
```

I am open for any hint.

Reply:
The base class is abstract, isn't it?

Reply:
Yes, it is, why? Sorry I cannot follow...

Reply:
@hoegertn LoadBalancedFargateService needs a vpc or a cluster with vpc specified, but it doesn't actually require capacity, where as a LoadBalancedEc2Service needs capacity which requires a cluster to be present with capacity in it.

Reply:
Can you ensure that for EC2, cluster is required?

Reply:
I can try to implement this, but I do not really see why. It is currently perfectly fine to add a cluster without capacity and during synth time it then fails validation. Without specifying a cluster this is the same behavior. As the cluster property is public you can add capacity like `service.cluster.addCapacity(...)`

Reply:
You can actually use [hasEc2Capacity](https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-ecs/lib/cluster.ts#L199) in cluster to check if capacity has been added to fail the build rather than wait till synthesis time.

Reply:
I am with @hoegertn. Why not let people create the service and then add the capacity to the cluster? I can totally see a use case for such a pattern for people. This is a classic example of "late validation".

Reply:
I don't disagree that it would work, but in the case of an Ec2 service, this pattern would be incomplete. You'd essentially allow someone to construct this pattern without actually providing the mandatory minimum properties.

Reply:
This is the same when you provide a cluster but do not add capacity to it ...

Reply:
@hoegertn is correct. We have other similar cases for “late validation”, like codepipeline (you can create a pipeline with no stages and then add them, but at least one is required). This essentially the only reason we have the “validate” hook.

Reply:
> @hoegertn is correct. We have other similar cases for “late validation”, like codepipeline (you can create a pipeline with no stages and then add them, but at least one is required). This essentially the only reason we have the “validate” hook.

@eladb the intent behind the constructs in ecs-patterns is to provide an opinionated approach to creating ecs services following what is considered best practices. I definitely agree that this works, but by relying on late validation, and not requiring capacity to be an input prop, we're not providing an opinion (allowing the user to decide if they want to add capacity when we know that they need to provide capacity), we're straying away from the intentions of the ecs-patterns module. @SoManyHs thoughts?

I would suggest that AddCapacityOptions (if cluster is not provided) or cluster with capacity already added to it should be required.

Reply:
While I agree with this opinion I do not see how this PR changes the current behavior as it currently possible to add clusters without capacity. So I think we could have this PR as is and create an extra issue about late vs. instant validation of capacity whether providing a cluster or not. Any thoughts?

Reply:
I'm okay with this.